docs/RELEASE.md

```diff
@@ -0,0 +1,79 @@
+# Release
+
+This document aims to describe how to perform a release of the Horreum python library.
+
+The versioning pattern should follow the [Horreum](https://github.com/Hyperfoil/Horreum) 
+versioning scheme to keep coherence among all Horreum-related projects versions.
+
+## Procedure
+
+### Tag a new version
+
+Checkout to your branch, either a `stable` (e.g., `0.12.x`) or the `main` one.
+
+```bash
+git checkout origin/main
+```
+
+Update the project version:
+
+```bash
+poetry version patch
+```
+
+This will bump your version, from `0.12-dev` to `0.12`.
+
+To double-check the version, run:
+```bash
+poetry version                                                                                                                                                      [15:55:39]
+# horreum 0.12
+```
+
+Commit the changes and tag a new version:
+```bash
+git add .
+git commit -m "Tag version 0.12"
+git tag v0.12
+```
+
+Ensure the tag is in the form of `v$(poetry version)`.
+
+Push changes and tag:
+```bash
+git push origin main
+git push origin v0.12
+```
```

Comment:
I agree that the overall release process can be improved and automated, but right now I would keep it as simple as possible (at least until we have a first package release) then we can improve it.

In the future all _pushes to main_ could be converted to automated pull request creation, already implemented similar approach in the past but I think it is at lower priority atm.

Reply:
> What is weird here is that this procedure implies that creating a release requires changes to the main branch. it really shouldn't -- you should be able to create a release from any point on main

Well ithe procedure is actually describe the most common scenario where the tag is created from `main` which should be the best practice (unless you are release a bug fixing releas from a stable branch, in that case you just need to tag from `0.12x` or whatever)

The commit on main is just to keep the version coherent with the tag, and then update to the next development version. (Very similar approach to what Horreum does).

Anyway I think that this process can be optimized whenever I will get chance to really perform a first release/tag. I wouldn't spend much time in documenting something that has been never tried concretely.

Reply:
In my experience, it is often the case that the `main` branch has progressed beyond the point where a release branch is created when the release is cut.  This might not be the case for this repo, which hopefully won't see a lot of active development once everything is working.

In most of the projects that I've worked on, the `main` branch was called "development" or "V0" or something else which was clearly "not for release", and anything that we wanted to share came from a tag or branch which was suitably labeled ("v<next>-alpha" or something).  So, creating a release was wholly independent of the activities on the `main` branch.  Things like setting the release name were done on the release branch (hopefully in its the initial commit... 🤣).